### ISSUE TITLE
[BUG] TimeSeriesDataFrame.train_test_split produces incorrect split when unsorted in time

### ISSUE THREAD

**Issue by JanusAsmussen** [autogluon/autogluon#4257]
**Bug Report Checklist**

- [x] I provided code that demonstrates a minimal reproducible example.  
- [x] I confirmed bug exists on the latest mainline of AutoGluon via source install.  
- [x] I confirmed bug exists on the latest stable version of AutoGluon.  

**Describe the bug**
The `TimeSeriesDataFrame.train_test_split` method produces train sets which may include test set timestamps when the dataframe is unsorted in time.

**Expected behavior**
I'd expect the method to either raise an error when the dataframe is unsorted in time or automatically sort the dataframe to ensure a valid split. The split should be robust to the order of the data to prevent any risk of temporal information leakage.

**To Reproduce**
```python
import pandas
from autogluon.timeseries import TimeSeriesDataFrame

data = pandas.DataFrame(
    {
        "item_id": [0, 0, 0, 1, 1, 1, 2, 2, 2],
        "timestamp": [
            "2019-01-01",
            "2019-01-02",
            "2019-01-03",
            "2019-01-02",
            "2019-01-03",
            "2019-01-04",
            "2019-01-03",
            "2019-01-04",
            "2019-01-05",
        ],
        "target": [0, 1, 2, 3, 4, 5, 6, 7, 8],
    }
)

autogluon_dataframe = TimeSeriesDataFrame.from_data_frame(
    df=data,
    id_column="item_id",
    timestamp_column="timestamp",
)

# Checks if the maximum timestamp in train is less than the maximum timestamp in test (PASSES)
sorted_train, sorted_test = autogluon_dataframe.sort_index().train_test_split(prediction_length=1)
assert (sorted_train.reset_index().groupby("item_id").timestamp.max() < sorted_test.reset_index().groupby("item_id").timestamp.max()).all()

# # Checks if the maximum timestamp in train is less than the maximum timestamp in test (FAILS)
shuffled_train, shuffled_test = autogluon_dataframe.sample(frac=1, random_state=42).train_test_split(prediction_length=1)
assert (shuffled_train.reset_index().groupby("item_id").timestamp.max() < shuffled_test.reset_index().groupby("item_id").timestamp.max()).all()
```

**Installed Versions**
The bug exists in both mainline and stable versions. Beneath, the dependencies for mainline.
<details>

```python
INSTALLED VERSIONS
------------------
date                   : 2024-06-11
time                   : 17:46:11.958736
python                 : 3.11.2.final.0
OS                     : Darwin
OS-release             : 23.5.0
Version                : Darwin Kernel Version 23.5.0: Wed May  1 20:14:38 PDT 2024; root:xnu-10063.121.3~5/RELEASE_ARM64_T6020
machine                : arm64
processor              : arm
num_cores              : 12
cpu_ram_mb             : 32768.0
cuda version           : None
num_gpus               : 0
gpu_ram_mb             : []
avail_disk_size_mb     : 638561
accelerate             : 0.21.0
autogluon              : 1.1.0
autogluon.common       : 1.1.1b20240611
autogluon.core         : 1.1.1b20240611
autogluon.features     : 1.1.1b20240611
autogluon.multimodal   : 1.1.0
autogluon.tabular      : 1.1.1b20240611
autogluon.timeseries   : 1.1.1b20240611
boto3                  : 1.34.79
catboost               : None
defusedxml             : 0.7.1
evaluate               : 0.4.1
fastai                 : 2.7.14
gluonts                : 0.15.1
hyperopt               : 0.2.7
imodels                : None
jinja2                 : 3.1.2
joblib                 : 1.3.2
jsonschema             : 4.21.1
lightgbm               : 4.3.0
lightning              : 2.2.5
matplotlib             : 3.8.2
mlforecast             : 0.10.0
networkx               : 3.1
nlpaug                 : 1.1.11
nltk                   : 3.8.1
nptyping               : 2.4.1
numpy                  : 1.26.4
nvidia-ml-py3          : 7.352.0
omegaconf              : 2.2.3
onnxruntime-gpu        : None
openmim                : 0.3.9
optimum                : 1.18.0
optimum-intel          : 1.16.0
orjson                 : 3.10.0
pandas                 : 2.1.4
pdf2image              : 1.17.0
Pillow                 : 10.1.0
psutil                 : 5.9.8
pytesseract            : 0.3.10
pytorch-lightning      : 2.2.5
pytorch-metric-learning: 2.3.0
ray                    : 2.10.0
requests               : 2.31.0
scikit-image           : 0.20.0
scikit-learn           : 1.4.0
scikit-learn-intelex   : None
scipy                  : 1.12.0
seqeval                : 1.2.2
setuptools             : 65.5.1
skl2onnx               : None
statsforecast          : 1.4.0
tabpfn                 : None
tensorboard            : 2.16.2
text-unidecode         : 1.3
timm                   : 0.9.16
torch                  : 2.3.1
torchmetrics           : 1.2.1
torchvision            : 0.16.2
tqdm                   : 4.66.1
transformers           : 4.38.2
utilsforecast          : 0.0.10
vowpalwabbit           : None
xgboost                : 2.0.3
```

</details>

**Comment by shchur** [comment]
@JanusAsmussen, thank you for a detailed bug report! I have created a PR to fix this issue (#4259), we will include it in the v1.2.0 release since it's a potentially breaking change (the behavior may change if `item_id` is not sorted alphabetically).

**Comment by JanusAsmussen** [comment]
@shchur, no problem - thanks for the quick resolution.
